napari/utils/transforms/transform_utils.py

```diff
@@ -215,3 +230,66 @@ def shear_matrix_from_angle(angle, ndim=3, axes=(-1, 0)):
     matrix = np.eye(ndim)
     matrix[axes] = np.tan(np.deg2rad(90 - angle))
     return matrix
+
+
+def is_upper_triangular(matrix):
+    """Check if a matrix is upper triangular.
+
+    Parameters
+    ----------
+    matrix : np.ndarray
+        Matrix to be checked.
+
+    Returns
+    -------
+    bool
+        Whether matrix is upper triangular or not.
+    """
+    return np.allclose(matrix, np.triu(matrix))
+
+
+def is_lower_triangular(matrix):
+    """Check if a matrix is lower triangular.
+
+    Parameters
+    ----------
+    matrix : np.ndarray
+        Matrix to be checked.
+
+    Returns
+    -------
+    bool
+        Whether matrix is lower triangular or not.
+    """
+    return np.allclose(matrix, np.tril(matrix))
+
+
+def check_shear_triangular(matrix):
```

Comment:
```suggestion
def check_matrix_triangular(matrix):
```

I think this function isn't specific to shear, so we should give it a more general name.